share/availability/discovery/discovery.go

```diff
@@ -169,3 +169,18 @@ func (d *Discovery) Advertise(ctx context.Context) {
 		}
 	}
 }
+
+// Peers provides a list of discovered peers in the "full" topic.
+// If Discovery hasn't found any peers, it blocks until at least one peer is found.
+func (d *Discovery) Peers(ctx context.Context) ([]peer.ID, error) {
+	if d.set.Size() > 0 {
+		return d.set.Peers(), nil
+	}
+
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	case p := <-d.set.waitPeer:
```

Comment:
nit: this break encapsulation of the set and putting this logic to the set itself would be cleaner.

Reply:
yeah, I know, thanks.